libs/blocks/tabs/tabs.js

```diff
@@ -64,52 +53,106 @@ function changeTabs(e) {
 }
 
+function configTabs(config) {
+  if(config['active-tab']) {
+    const id = `tab-${config['tab-id']}-${getStringKeyName(config['active-tab'])}`;
+    const sel = document.getElementById(id);
+    if(sel) sel.click();
+  }
+}
+
+const isElementInContainerView = (targetEl) => {
+  const rect = targetEl.getBoundingClientRect();
+  return (
+    rect.top >= 0 &&
+    rect.left >= 0 &&
+    rect.bottom <= (window.innerHeight || /* c8 ignore next */ document.documentElement.clientHeight) &&
+    rect.right <= (window.innerWidth || /* c8 ignore next */ document.documentElement.clientWidth)
+  );
+};
+
+const scrollTabIntoView = (e) => {
+  const isElInView = isElementInContainerView(e);
+  /* c8 ignore next */
+  if (!isElInView) e.scrollIntoView({ behavior: 'smooth', block: 'nearest', inline: 'center' });
+};
+
 let initCount = 0;
-const init = (element) => {
-  const rows = element.querySelectorAll(':scope > div');
+const init = (e) => {
+  const rows = e.querySelectorAll(':scope > div');
   /* c8 ignore next */
   if(!rows.length) return;
-  const tabList = createTag('div', {role: 'tablist'});
-  const tabListContainer = createTag('div', {class: 'tabList-container container'});
-  const containerWrapper = createTag('div', {class: 'container'});
-  const tabContentContainer = createTag('div', {class: 'tabContent-container'}, containerWrapper);
-  let btnClass = [...element.classList].includes('quiet') ? 'heading-XS' : 'heading-XS'; // tabList size
-  let singleColRows = 0;
-  rows.forEach((row) => { if (row.childElementCount === 1) singleColRows += 1; });
-  if (singleColRows) {
-    const rowHeadler = rows[0].querySelector('h1, h2, h3, h4, h5, h6');
-    rows[0].classList.add('tab-headline', 'container');
-    tabList.setAttribute('aria-label', rowHeadler.textContent);
+
+  // Tab Content
+  const tabContentContainer = createTag('div', {class: 'container'});
+  const tabContent = createTag('div', {class: 'tabContent'}, tabContentContainer);
+  e.append(tabContent);
+
+  // Tab List
+  const tabList = rows[0];
+  const tabId = `tabs-${initCount}`;
+  e.id = tabId;
+  tabList.classList.add('tabList');
+  tabList.setAttribute('role', 'tablist');
+  const tabListContainer = tabList.querySelector(':scope > div');
+  tabListContainer.classList.add('tabList-container');
+  const tabListItems = rows[0].querySelectorAll(':scope li');
+  if (tabListItems) {
+    let btnClass = [...e.classList].includes('quiet') ? 'heading-XS' : 'heading-XS'; // tabList size
+    tabListItems.forEach((item, i) => {
+      const tabName = getStringKeyName(item.textContent);
+      const tabBtnAttributes = {
+        role: 'tab',
+        class: btnClass,
+        id: `tab-${initCount}-${tabName}`,
+        tabindex: (i > 0) ? '0' : '-1',
+      }
+      const tabBtn = createTag('button', tabBtnAttributes);
+      tabBtn.setAttribute('aria-selected', (i === 0) ? 'true' : 'false');
+      tabBtn.setAttribute('aria-controls', `tab-panel-${initCount}-${tabName}`);
```

Comment:
Doesn't `createTag` use `setAttribute` as well? Why not add them to the attributes object?

Reply:
@meganthecoder - I tried that, but It doesn't seem to work w/ hyphenated attribute names. :( maybe theres a way?

Reply:
I tried it and it seemed to work fine. Also you're missing a semi-colon. Do you have the ESLint extension set up?
```
     const tabBtnAttributes = {
        role: 'tab',
        class: btnClass,
        id: `tab-${initCount}-${tabName}`,
        tabindex: (i > 0) ? '0' : '-1',
        'aria-selected': (i === 0) ? 'true' : 'false',
        'aria-controls': `tab-panel-${initCount}-${tabName}`,
      };
```

Reply:
Ok, thanks. I fixed that. I must have had tried incorrect syntax. Im lookin into my eslint IDE setting. Thanks